### PR TITLE
Intercepted tx changed trace print

### PR DESCRIPTION
**Pull request by iulianpascalau**
- removed unnecessary transactions request when a miniblock was received
- removed the second marshaling of reward txs
- added valuable prints in interceptors and p2p messenger (originator pid and originator counter)
- added prints that periodically output how many peers are connected/known by the host

### CODE REVIEW

File: process/block/preprocess/rewardsHandler.go [hunk 1]
```diff
@@ -245,25 +245,8 @@ func (rtxh *rewardsHandler) CreateBlockStarted() {
...
-
-		marshaledTx, err := rtxh.marshalizer.Marshal(rTx)
-		if err != nil {
-			return nil, process.ErrMarshalWithoutSuccess
-		}
-		marshaledTxs[idx] = marshaledTx
-	}
-
-	return marshaledTxs, nil
+func (rtxh *rewardsHandler) CreateMarshalizedData(_ [][]byte) ([][]byte, error) {
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply process/block/preprocess/rewardsHandler.go]: Not quite, as the code is needed for the other InterimProc implementations

[the remark this answers is not in the document]
Comment (iulianpascalau) [reply process/block/preprocess/rewardsHandler.go]: refactored

File: process/interface.go [hunk 2]
```diff
@@ -143,14 +143,18 @@ type IntermediateTransactionHandler interface {
...
-	CreateMarshalizedData(txHashes [][]byte) ([][]byte, error)
 	SaveCurrentIntermediateTxToStorage() error
 	GetAllCurrentFinishedTxs() map[string]data.TransactionHandler
 	CreateBlockStarted()
 	GetCreatedInShardMiniBlock() *block.MiniBlock
 	IsInterfaceNil() bool
 }
 
+// DataMarshalizer defines the behavior of a structure that is able to marshalize containing data
+type DataMarshalizer interface {
```
Comment (raduchis) [inline process/interface.go]: :+1:

File: p2p/p2p.go [hunk 3]
```diff
@@ -198,3 +201,21 @@ type PeerDiscoveryFactory interface {
...
+	prettyPid := msg.Peer().Pretty()
+	lenPrettyPid := len(prettyPid)
+	if lenPrettyPid > displayLastPidChars {
+		return "..." + prettyPid[lenPrettyPid-displayLastPidChars:]
+	}
+
+	return prettyPid
+}
+
+// MessageOriginatorSeq will output the sequence number if it can be converted to an uint64, otherwise it will
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply p2p/p2p.go]: right

[the remark this answers is not in the document]
Comment (iulianpascalau) [reply p2p/p2p.go]: done

**Pull request merged**